lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,1608 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: cloudfront_distribution
+short_description: create, update, delete, duplicate and validate aws cloudfront distributions
+description:
+    - Allows for easy creation, updating, deletion, duplication and validation of CloudFront distributions.
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+version_added: "2.4"
+author: Willem van Ketwich (@wilvk)
+options:
+    state:
+      description:
+        - The desired state of the distribution or streaming distribution.
+          present - creates a new distribution
+          absent - deletes an existing distribution
+          updated - updates an existing distribution
+          duplicated - duplicates an existing distribution
```

Comment:
No worries - will remove. The idea was if a user has a base distribution with all their finely configured cache behaviours and want to duplicate it for a new website in the creation process, this would automate that. However, if they really want to duplicate a distribution they can just get the config from cloudfront_facts and use that as a basis anyway.